contractcourt/briefcase.go

```diff
@@ -802,6 +781,59 @@ func (b *boltArbitratorLog) FetchChainActions() (ChainActionMap, error) {
 	return actionsMap, nil
 }
 
+// InsertConfirmedCommitSet stores the known commitment set at the time of
+// channel closure. We'll use this to reconstruct our set of chain actions anew
+// based on the confirmed and pending commitment state.
+//
+// NOTE: Part of the ContractResolver interface.
+func (b *boltArbitratorLog) InsertConfirmedCommitSet(commits *CommitSet) error {
+	return b.db.Update(func(tx *bbolt.Tx) error {
+		scopeBucket, err := tx.CreateBucketIfNotExists(b.scopeKey[:])
+		if err != nil {
+			return err
+		}
+
+		var b bytes.Buffer
+		if err := encodeCommitSet(&b, commits); err != nil {
+			return err
+		}
+
+		return scopeBucket.Put(commitSetKey, b.Bytes())
```

Comment:
It still feels like redundant data. The commit sets are already persisted by the channel and now here again. Couldn't storing just the type of commit (local, remote, remote pending) be enough?

Reply:
Once the channel has been marked as pending closed, we no longer have access to that data since the channel close summary doesn't store the full commitment information. Once this is written, you're correct that the data is redundant, but once we proceed to the next call to mark the channel as pending close, it's the only area in the database that we store this information. Looking back upon it, it was a mistake to remove all the channel state once we moved from open -> pending close. A few times now, we've had to supplement the pending close information in order to carry out tasks like what's implemented in this PR.

Reply:
So shouldn't that data be kept a bit longer in the channel buckets then? To prevent serializing them there when the channel is open and storing it in the new store when it is pending close?

Reply:
It's possible for us to head back in this direction, but it's a larger change over all. We'd either need to move all that state into the current channel close summary, or modify `CloseChannel` to hold onto all the state or "move" all the channel state into a different bucket since if it lingers there, then many other sub-systems/callers will think the channel is still open.

Reply:
If it touches many other sub-systems, I can imagine it is a bigger change. It seems to me that only persisting commit txes in the channel state (where they already are stored) is a cleaner solution, but probably to big of a refactoring for now.